src/ws/cockpitauth.c

```diff
@@ -668,143 +579,147 @@ create_creds_for_spawn_authenticated (CockpitAuth *self,
+on_auth_process_close (CockpitAuthProcess *auth_process,
+                       GError *error,
+                       const gchar *problem,
+                       gpointer user_data)
 {
   AuthData *ad = user_data;
   /* Only report errors */
   if (error || ad->pending_result)
     auth_data_complete_result (ad, error);
 }
 
+static AuthData *
+create_auth_data (CockpitAuth *self,
+                  const gchar *name,
+                  const gchar *application,
+                  const gchar *type,
+                  const gchar *remote_peer,
+                  const gchar *logname,
+                  GBytes *input)
+{
+  AuthData *ad = NULL;
+  ad = g_new0 (AuthData, 1);
+  ad->refs = 1;
+  ad->id = cockpit_auth_nonce (self);
+  ad->pending_result = NULL;
+  ad->response_data = NULL;
+  ad->tag = NULL;
+  ad->remote_peer = g_strdup (remote_peer);
+  ad->auth_type = g_strdup (type);
+  ad->authorization = g_bytes_ref (input);
+  ad->application = g_strdup (application);
+
+  ad->auth_process = g_object_new (COCKPIT_TYPE_AUTH_PROCESS,
+                                   "pipe-timeout", timeout_option ("timeout", type,
+                                                                cockpit_ws_auth_process_timeout),
+                                   "idle-timeout", timeout_option ("response-timeout", type,
+                                                                cockpit_ws_auth_response_timeout),
+                                   "id", ad->id,
+                                   "logname", logname,
+                                   "name", name,
+                                   /* The wanted authfd for this command, default is 3 */
+                                   "wanted-auth-fd", cockpit_conf_guint (type, "authFD", 3, 1024, 3),
+                                   NULL);
+  return ad;
+}
+
+static void
+start_auth_process (CockpitAuth *self,
+                    AuthData *ad,
+                    GBytes *auth_data,
+                    GSimpleAsyncResult *result,
+                    gpointer tag,
+                    const gchar **argv)
+{
+  GError *error = NULL;
+
+  g_simple_async_result_set_op_res_gpointer (result,
+                                             auth_data_ref (ad), auth_data_unref);
+
+  if (cockpit_auth_process_start (ad->auth_process, argv, -1, FALSE, &error))
+    {
+      auth_data_add_pending_result (ad, result);
+      g_signal_connect (ad->auth_process, "message",
+                        G_CALLBACK (on_auth_process_message),
+                        ad);
+      g_signal_connect (ad->auth_process, "close",
+                        G_CALLBACK (on_auth_process_close),
+                        ad);
+      g_signal_connect (ad->auth_process, "close",
+                        G_CALLBACK (purge_auth_id),
+                        self);
+      cockpit_auth_process_write_auth_data (ad->auth_process, auth_data);
+    }
+  else
+    {
+      g_warning ("failed to start %s: %s", argv[0], error->message);
```

Comment:
This is hard to trigger. We close all the parent fds in the child setup call back, that has the side effect of closing the way that the grandchild process bubbles errors back up, that's not to say this can never happen, there are classes of errors like resource errors that could trigger this, but those are hard to trigger in a unit test. The easy way of triggering this ie a invalid path, doesn't work.
BTW this same behavior can be seen on current master.